### PR TITLE
Opencv protobuf

### PR DESCRIPTION
**Pull request by bvanessen**
Added a dependency on protobuf when building version 3.3.0

### CODE REVIEW

File: var/spack/repos/builtin/packages/opencv/package.py [hunk 1]
```diff
@@ -90,6 +90,7 @@ class Opencv(CMakePackage):
     depends_on('qt', when='+qt')
     depends_on('java', when='+java')
     depends_on('py-numpy', when='+python', type=('build', 'run'))
+    depends_on('protobuf@3.1.0', when='@3.3.0:')
```
Comment (becker33) [inline var/spack/repos/builtin/packages/opencv/package.py]: Should this be dependent on `protobuf@3.1.0` or `protobuf@3.1.0:`?

Comment (bvanessen) [reply var/spack/repos/builtin/packages/opencv/package.py]: Yes, their code does not compile with later versions of protobuf.

File: var/spack/repos/builtin/packages/opencv/package.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ class Opencv(CMakePackage):
     variant('java', default=False,
             description='Activates support for Java')
     variant('openmp', default=False, description='Activates support for OpenMP threads')
-    variant('core', default=False, description='Include opencv_core module into the OpenCV build')
+    variant('core', default=True, description='Include opencv_core module into the OpenCV build')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/opencv/package.py]: I'm curious why you changed the default to `True` here - do you think most users would want this enabled? Or is it not building for you when `core=False`?

**Comment by bvanessen** [comment]
It does not build for people when it is set to false.


Brian C. Van Essen

vanessen1@llnl.gov<mailto:vanessen1@llnl.gov>

(w) 925-422-9300

(c) 925-290-5470

---

Sent from my iPhone

On Sep 20, 2017, at 7:32 PM, scheibelp <notifications@github.com<mailto:notifications@github.com>> wrote:


@scheibelp commented on this pull request.

________________________________

In var/spack/repos/builtin/packages/opencv/package.py<https://github.com/LLNL/spack/pull/5232#discussion_r140137363>:

> @@ -68,7 +68,7 @@ class Opencv(CMakePackage):
     variant('java', default=False,
             description='Activates support for Java')
     variant('openmp', default=False, description='Activates support for OpenMP threads')
-    variant('core', default=False, description='Include opencv_core module into the OpenCV build')
+    variant('core', default=True, description='Include opencv_core module into the OpenCV build')


I'm curious why you changed the default to True here - do you think most users would want this enabled? Or is it not building for you when core=False?

**Comment by scheibelp** [comment]
> It does not build for people when it is set to false.

Got it, I didn't realize this was addressing a specific issue (https://github.com/LLNL/spack/issues/5159) until I read the commit messages.

Thanks!

**Pull request merged**